src/nebula/Job/Runner/JobStatisticsCalculator.cs

```diff
@@ -163,13 +163,18 @@ public void ReportProcessorInvocationException(Exception exception)
         private void ReportException(string source, Exception exception)
         {
             var message = $"{source}: {exception.Message}";
-            if (exception.InnerException != null)
+            
+            while (exception?.InnerException != null)
+            {
                 message += Environment.NewLine + exception.InnerException.Message;
-
+                exception = exception.InnerException;
+            }
+            
             _jobStore.AddException(_tenantId, _jobId, new JobStatusErrorData
                 {
                     ErrorMessage = message,
-                    Timestamp = DateTime.UtcNow.Ticks
+                    Timestamp = DateTime.UtcNow.Ticks,
+                    StackTrace = exception.StackTrace
```

Comment:
I'm not sure if getting the stack trace could, in some cases, result in an exception itself or not.
I propose wrapping the whole method in a try-catch to avoid any unhandled exceptions from the exception reporting itself.